### PR TITLE
Depend on tokenbridge contracts

### PR DESCRIPTION
**Pull request by fleupold**
This PR is pulling in the [xDAI tokenbridge repo](https://github.com/poanetwork/tokenbridge-contracts) as a submodule. This is needed so that we can build a custom version of xOWL that is based on their standard bridge contract (`PermittableToken.sol`). They currently don't expose their contracts as an npm module and thus using a submodule here seems most pragmatic to me.

Also their contracts require solidity 0.4.24 which is why I'm adding the relevant config changes in this PR (so that the next PR can focus on the smart contract code + test)

Happy to hear other suggestions.

### Test Plan

Making sure we can lint and compile everything without errors.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+vendor/**/*.js
```
Comment (bh2smith) [inline .eslintignore:1]:
```suggestion
vendor/**/*.js

```

**Review by bh2smith — APPROVED** [review]
Looks pretty good.

File: package.json [hunk 2]
```diff
@@ -13,9 +13,9 @@
     "networks-inject-dep": "node src/inject_network_info_deps.js",
     "networks-reset": "truffle networks --clean && npm run networks-inject",
     "networks": "truffle networks",
-    "compile-todo": "eslint --fix . && eslint . && truffle compile",
-    "compile": "truffle compile",
-    "restore": "truffle compile --all && npm run networks-reset",
+    "compile-todo": "eslint --fix . && eslint . && yarn compile",
+    "compile": "truffle compile -all && truffle compile --all --config truffle.4.js",
```
Comment (nlordell) [inline package.json:17]: Is there a missing `-`?

```suggestion
    "compile": "truffle compile --all && truffle compile --all --config truffle.4.js",
```

**Review by nlordell — APPROVED** [review]

**Review by josojo — APPROVED** [review]
Yeah, I also don't have a better suggestion.

Would be great to see the tests run with 0.4 an 0.5 contracts within Travis, so that we always have a confirmation that both versions work.

**Pull request merged**